### ISSUE TITLE
Implement GET `/proposals/{proposalId}` endpoint

### ISSUE THREAD

**Issue by slifty** [PhilanthropyDataCommons/service#101]
As mentioned in #34 we need the ability to load a specific proposal.  We have explored the pattern of populating deep fields when loading a single ID and I think it would be appropriate to do so here as well, which would mean the returned proposal would include the various proposal versions and proposal field values.

**Comment by bickelj** [comment]
Had we originally decided to return deep objects on all the single-id endpoints? Or should we default to shallow objects on all endpoints and then offer deep objects on some, as suggested around https://github.com/PhilanthropyDataCommons/service/pull/157#issuecomment-1349388808? PR #192 assumes shallow objects to be the default. The next step would be to allow deep objects when specified.

**Comment by bickelj** [comment]
I am still working on the deep objects code and tests. But here is an example response for when asking to include fields and values with `curl ... http://host/proposals/1?includeFieldsAndValues=true | jq`:
```json
{
  "id": 1,
  "applicantId": 1,
  "opportunityId": 1,
  "externalId": "AnIdGeneratedByAGms",
  "createdAt": "2022-11-10T20:07:04.695Z",
  "versions": [
    {
      "id": 2,
      "proposalId": 1,
      "applicationFormId": 4,
      "version": 1,
      "createdAt": "2023-01-04T16:28:47.276Z",
      "fieldValues": [
        {
          "id": 1,
          "proposalVersionId": 2,
          "applicationFormFieldId": 3,
          "position": 1,
          "value": "Jones",
          "createdAt": "2023-01-04T16:29:57.877Z",
          "applicationFormField": {
            "id": 3,
            "applicationFormId": 4,
            "canonicalFieldId": 280,
            "position": 1,
            "label": "Family Name",
            "createdAt": "2023-01-04T16:27:40.175Z"
          }
        },
        {
          "id": 2,
          "proposalVersionId": 2,
          "applicationFormFieldId": 4,
          "position": 2,
          "value": "Jim",
          "createdAt": "2023-01-04T16:30:08.974Z",
          "applicationFormField": {
            "id": 4,
            "applicationFormId": 4,
            "canonicalFieldId": 279,
            "position": 2,
            "label": "Given Name",
            "createdAt": "2023-01-04T16:27:40.175Z"
          }
        }
      ]
    }
  ]
}
```

**Comment by bickelj** [comment]
Yesterday I got a draft of the alternative approach at https://github.com/PhilanthropyDataCommons/service/pull/198#issuecomment-1384735956 working locally, although it is in a less-than-elegant state right now. My goal is to clean it up a bit today. There will need to be a fourth query to include the application form fields that relate to the proposal values, too.

**Comment by bickelj** [comment]
There are two more things on the /proposals endpoints before I am comfortable closing this issue.
1. Docs updates
2. If `throw error` inside a `catch` inside a `then` does not propagate to the outer promise chain, we need to call `next` inside the innermost `catch` blocks.

**Comment by bickelj** [comment]
Resolved by merge of #192, #210, and #232.
